ecms/src/main/java/org/exoplatform/platform/qa/ui/ecms/pageobject/SiteExplorerHome.java

```diff
@@ -1601,4 +1604,39 @@ public void verifyWebContentInformationCreatedSuccessfully(String Content){
   public enum defineValueBehavior {
     CREATE_NEW, REMOVE_EXISTING, REPLACE_EXISTING, THROW_EXEPTION;
   }
+
+
+  /**
+   * Click on View Document button
+   */
+  public void clickViewDocumentButton() {
+    $(byText("View Document")).waitUntil(Condition.visible,Configuration.openBrowserTimeoutMs).click();
+  }
+
+
+  /**
+   * Click on Edit Online button
+   */
+  public void clickEditOnine() {
+    $(byText("Edit Online")).click();
+  }
+
+  /**
+   * Click on Delete button
+   */
+  public void checkButtonDocument(String officeDocument) {
+    info("click on CheckBox button for Document");
+    $(byXpath("(//div[@class='columnText']//span[@class='nodeName' and text()='${word}']/preceding::span[@class='uiCheckbox'])[last()]".replace("${word}",officeDocument))).waitUntil(Condition.visible,Configuration.timeout).click();
```

Comment:
Elements should not be selected by xpath, but rather by id or classes.

Reply:
the data is entered in string type, so it will not affect the Xpath

Reply:
Yes but XPath is fragile, since the DOM can often change. That's why it is better to rely on id or classes.

Reply:
all the elements are defined with class, only the name of the document to upload

Reply:
Yes but the test will fail if we change a span to a div for example. By using only id and classes, the test is more robust.